core/src/main/java/io/questdb/cutlass/line/array/AbstractArray.java

```diff
@@ -94,6 +134,61 @@ public void close() {
         closed = true;
     }
 
+    /**
+     * Reshapes the array to the specified dimensions.
+     * <p>This method allows changing the array's shape after creation, enabling
+     * dynamic resizing based on runtime requirements.
+     * <br>
+     * Note: It resets the append position to the start of the array.
+     *
+     * @param shape the new dimensions for the array
+     * @throws IllegalStateException    if the array is already closed
+     * @throws IllegalArgumentException if shape has invalid dimensions
```

Comment:
this documentation is incorrect, `array.setDimLen()` is liable to throw CairoException - can we perhaps standardise the exception handling ?